core/subsystems/api.py

```diff
@@ -37,54 +38,77 @@ def resolve_session_func(self, session):
 
 class API:
     """A class for interacting with the ingame and web based APIs."""
+
     def __init__(self, config):
+        self._logger = logging.getLogger(__name__)
         if not config:
             raise ApiError("No config given to constructor.", "__init__")
 
-        self._auth = ""
-
+        self._api_auth = ""
         if config.api["auth"]:
-            self._auth = config.api["auth"]
+            self._api_auth = config.api["auth"]
 
         self._api_path = ""
-
         if config.api["path"]:
             self._api_path = config.api["path"]
 
+        self._forum_auth = ""
+        if config.forum["auth"]:
+            self._forum_auth = config.forum["auth"]
+
+        self._forum_path = ""
+        if config.forum["path"]:
+            self._forum_path = config.forum["path"]
+
         self._server_host = config.server["host"]
         self._server_port = config.server["port"]
         self._server_auth = config.server["auth"]
 
         self._monitor_host = config.monitor["host"]
         self._monitor_port = config.monitor["port"]
 
-    async def query_web(self, uri, method, data=None, return_keys=None,
-                        enforce_return_keys=False):
+    async def query_web(self, uri, method, data={}, return_keys=None,
+                        enforce_return_keys=False, api_dest="api"):
         """
         A method for querying the home API of the bot.
         """
-        if not self._api_path:
+        valid_targets = ["forum", "api"]
+
+        if api_dest not in valid_targets:
+            self._logger.error("Invalid API Destination specified.")
             raise ApiError("No API path specified.", "query_web")
 
         if not uri:
+            self._logger.error("No URI sent.")
             raise ApiError("No URI sent.", "query_web")
 
         if not isinstance(method, ApiMethods):
+            self._logger.error("Bad method sent.")
             raise ApiError("Bad method sent.", "query_web")
 
-        dest = self._api_path + uri
-        use_headers = (method is not ApiMethods.GET)
-
-        if not data:
-            data = {}
-
-        data["auth_key"] = self._auth
-        arg_dict = {"url": dest}
+        # Validate a different dataset based on api_dest
+        arg_dict = {"url": None, "data": {}, "params": {}}
+        error_message_key = "error_msg"
+        if api_dest == "api":
+            if not self._api_path:
+                self._logger.error("No API path specified.")
+                raise ApiError("No API path specified.", "query_web")
+            arg_dict["url"] = self._api_path + uri
+            data["auth_key"] = self._api_auth
+
+        if api_dest == "forum":
```

Comment:
`elif` this and add an `else` clause that'll throw an `ApiError("Invalid API destination specified.")`